app/assets/scripts/components/slate/plugins/equation/equation-editor.js

```diff
@@ -104,43 +102,28 @@ export default function EquationEditor(props) {
     }
   }, [latexEquation]);
 
-  // Focus equation input when user clicks the preview.
-  const onEquationPreviewClick = useCallback(
-    (e) => {
-      // When the equation preview box gets clicked we want to focus the
-      // equation field except if the user is selecting the equation characters.
-      const node = equationBlockRef.current;
-      if (node) {
-        const display = node.querySelector('.katex-display');
-        if (display.contains(e.target)) return;
-      }
-
-      // When trying to select something in the editor with a selection
-      // elsewhere (for example the equation characters) the slate editor
-      // crashed. To fix this before doing anything we use the Selection api to
-      // clear the existing selection (which is not managed by slate) and then
-      // let slate handle this from a clean slate. Pun intended.
-      window.getSelection().removeAllRanges();
-      const path = ReactEditor.findPath(editor, element);
-      Transforms.select(editor, Editor.end(editor, path));
-    },
-    [editor, element]
-  );
+  const handleInputChange = (e) => setLatexEquation(e.target.value);
+  const handleSave = () => {
+    const path = element && ReactEditor.findPath(editor, element);
```

Comment:
Getting the node's path here so I remove and insert the equation on the same position inside `upsertEquation`.

Reply:
Using `ReactEditor` is the way to find a path to a node. It matches the element with the slate tree. Do you see any disadvantage with this method? Were you expecting a different behavior?

Instead of removing and adding you could just update the equation value, but I found that replacing a node is easier especially when the different types are not very compatible. I'd use the update function (Transforms.setNodes) when moving between compatible nodes - for example paragraph to lists or between list types.

Reply:
> Do you see any disadvantage with this method? Were you expecting a different behavior?

I don't think anything is wrong with this approach. It's just that the equation editor prescribes where in the document the equation should be placed. This introduces direct coupling between the equation editor and the rich-text editor, which I want to avoid in case we want to replace the equation editor in the future.

Reply:
This makes sense. Maybe you could pass a callback to the editor and handle it there?
Otherwise you could make the editor modal promise based and then capture the result (similar to the [confirmation prompt](https://github.com/NASA-IMPACT/nasa-apt-frontend/blob/develop/app/assets/scripts/components/common/confirmation-prompt.js), but perhaps this complicated things.)